Tests/Helpers/SharedMocks.swift

```diff
@@ -1,5 +1,38 @@
 @testable import RInAppMessaging
 
+class IAMPreference: UserInfoProvider {
```

Comment:
I know unit tests will take some time to (re)write and the final PR diff will be huge, but without unit tests we won't know if `UserInfoProvider` approach works.